File: govuknotify1/src/main/java/uk/gov/nationalarchives/tdr/govuknotify/NotifyEmailSenderProviderFactory.java
package uk.gov.nationalarchives.tdr.govuknotify;

import org.keycloak.email.EmailSenderProviderFactory;
import org.keycloak.Config.Scope;
import org.keycloak.email.EmailSenderProvider;
import org.keycloak.models.KeycloakSession;
import org.keycloak.models.KeycloakSessionFactory;
import uk.gov.service.notify.NotificationClient;

public class NotifyEmailSenderProviderFactory implements EmailSenderProviderFactory {
    @Override
    public EmailSenderProvider create(KeycloakSession session) {

        final String apiKey = "[api_key_value]";

        NotificationClient client = new NotificationClient(apiKey);

        return new NotifyEmailSenderProvider(client);
    }

    @Override
    public void init(Scope config) { }

    @Override
    public void postInit(KeycloakSessionFactory factory) { }

    @Override
    public void close() { }

    @Override
    public String getId() {

        return "govuknotify";
    }
}

File: govuk-notify-spi/src/main/scala/uk/gov/nationalarchives/notifyspi/NotifyEmailSenderProviderFactory.scala
package uk.gov.nationalarchives.notifyspi

import org.keycloak.Config
import org.keycloak.email.{EmailSenderProvider, EmailSenderProviderFactory}
import org.keycloak.models.{KeycloakSession, KeycloakSessionFactory}

class NotifyEmailSenderProviderFactory extends EmailSenderProviderFactory {

  private val emailSenderId = "govuknotify"

  override def create(session: KeycloakSession): EmailSenderProvider = {
    new NotifyEmailSenderProvider(sys.env)
  }

  override def init(config: Config.Scope): Unit = { }

  override def postInit(factory: KeycloakSessionFactory): Unit = { }

  override def close(): Unit = { }

  override def getId: String = {
    emailSenderId
  }
}
